app/src/main/java/com/example/pocketcloset/Fragments/BaseFragment.java

```diff
@@ -0,0 +1,77 @@
+package com.example.pocketcloset.Fragments;
+
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+
+import androidx.activity.result.contract.ActivityResultContracts;
+import androidx.core.content.FileProvider;
+import androidx.fragment.app.Fragment;
+
+import android.os.Environment;
+import android.provider.MediaStore;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.example.pocketcloset.R;
+
+import java.io.File;
+
+
+public class BaseFragment extends Fragment {
+    public static final String TAG ="BaseFragment";
+    public static final int CAPTURE_IMAGE_ACTIVITY_REQUEST_CODE = 42;
+    public File photoFile;
+    public String photoFileName = "photo.jpg";
+
+
+
+    void launchCamera() {
+
+        // create Intent to take a picture and return control to the calling application
+
+        Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+        // Create a File reference for future access
+        photoFile = getPhotoFileUri(photoFileName);
+        Log.i("BASE FRAG", photoFile.toString());
+
+        // wrap File object into a content provider
+        // required for API >= 24
+        // See https://guides.codepath.com/android/Sharing-Content-with-Intents#sharing-files-with-api-24-or-higher
+        Uri fileProvider = FileProvider.getUriForFile(getContext(), "com.codepath.fileprovider", photoFile);
+        intent.putExtra(MediaStore.EXTRA_OUTPUT, fileProvider);
+
+        // If you call startActivityForResult() using an intent that no app can handle, your app will crash.
+        // So as long as the result is not null, it's safe to use the intent.
+
+
+            if (intent.resolveActivity(getContext().getPackageManager()) != null) {
```

Comment:
nit: The indentation is inconsistent. Did the auto-format on save macro stop working?